src/app/meshes/sources.ts

```diff
@@ -30,7 +30,7 @@ export const sources = (api: KumaApi) => {
       const { size } = params
       const offset = params.size * (params.page - 1)
 
-      return api.getAllMeshInsights({ size, offset })
+      return api.getAllMeshes({ size, offset })
```

Comment:
**Comment**: The `/meshes` data source is used by OnboardingNotification.

Reply:
Q: What is the difference between the `total` property of the `meshes` endpoint and the `total` property of the `mesh-insights` endpoint? Do they both refer to the same piece of data? i.e. the total of meshes?

Reply:
Yes.

Reply:
Ok so, with that info, given that you are highly likely to land on the homepage that also uses `getAllMeshInsights`/`/mesh-insights` at the same time that the `OnboardingNotification` component makes its request, I would think it's a good idea to reuse that same single request instead of making two separate ones in order to get the same data? 

That being said, unless I'm missing something, I think the improvement here would be to leave everything as it is, and just make sure that the `size=`'s are the same, i.e. both should have `size=10`.

I think that would mean that in the majority of cases we only make 1 request.

Reply:
P.S. Ahhhh I remember now the OnboardingNotification currently has `size=50` as I wanted to match the it up with the original mesh listing back when I moved all of onboarding to use DataSource. This would have been previous to having `/me` and possibly before the smaller mesh listing on the homepage. 🤔 

Ok, this is starting to be clearer now. This isn't as clear cut as to what `size=` should be on the 3 DataSources involved here, but I still think they should all use the same `/mesh-insights` API endpoint. I'm kinda possibly thinking they should all use the `size` from `/me`.

I'm starting to think we should at least close this PR and possibly reopen another later, probably changing all 3 DataSources use `size=` from `/me`.

---

This is all dependent on me not having missed something here, let me know if so.

Reply:
Does DataSource, actually ensure this request is not duplicated? If we can’t ensure that, we’re over-fetching data for nothing.

Reply:
It doesn't 'ensure', it just makes it possible/likely.

It also caches, so if we use `/mesh-insights` on the OnboardingNotification, it means that if you then click on the mesh listing the cached data is already there as the 'loading state'.

Reply:
OnboardingNotification will be loaded on every page (i.e. also when someone refreshes the page) whereas the other two components are specific to their respective views. It’s basically chance that we sometimes avoid querying `/mesh-insights` twice plus it’s incidental that the home page happens to query `/mesh-insights`. Yes, it’s the most common home page, but now we started to call `/mesh-insights` on every page load instead of just where we need it.

I think we should load `/meshes` for this purpose. It’s the cheaper endpoint to query.

Reply:
Would it be fair to say that on any page below and `/meshes` if we get a 200 from the API we know that at least 1 mesh exists? (and for `/meshes` itself we query `/mesh-insights` so we can reuse that)

Reply:
I don’t understand the question. How is this assumption relevant to the issue at hand?

Reply:
Well in order to show the OnboardingNotification we firstly need to know if a mesh exists, which is why we make that API call that we are talking about. If we already know one exists, then we don't need to make it.

Reply:
Well, yes, but this component is also used in non-mesh pages so it doesn’t help us making any assumptions, right?

Reply:
Yeah for sure, but it sounds like a fair assumption to make is that if we land on any page below `/meshes` and its a 200 then we know a mesh exists.

So we are left with anything under browser URL `/zones` I think? I would guess you might be able to have a zone without having a mesh? Kinda unlikely, especially given we are talking about onboarding.. Actually 🤔  I'd say if you have a zone or a mesh then we shouldn't show the onboarding notification?

If that is the case we can probably say if we get a 200 from any page under `/meshes` or `/zones` then we know not to show the OnboardingNotification? If we land on `/gui` then we already use `/mesh-insights`, if we land on `/meshes` then we use already use  `/mesh-insights`. Does that sound correct?

Reply:
I think this is very much out of scope for this PR. We can of course consider changing the condition under which we show the onboarding notification, but it was never supposed to use the `/mesh-insights` API.

Reply:
Well I think it is very much in scope, and this is why I suggested closing this, or at the very least think it through, as I think we should think this through properly before making 2 requests for an incredibly common landing page rather than just the 1.

I think its important to understand that when I changed the code that you are changing back here, I did it whilst trying to reduce the amount of HTTP requests we make given we had a big problem with that. So 

>  it was never supposed to use the /mesh-insights API.

Yes, it was supposed to use the `/mesh-insights` API as I wanted to share the requests/cache.

Continuing to use `/mesh-insights` and sharing those requests means less requests for the incredibly common case. For the much less common case (and therefore slightly less important), we could improve in another differently scoped PR if you wanted to do that by omitting the request entirely if we get a 200 on deeper pages because we already know we have a mesh and/or zone.

> We can of course consider changing the condition under which we show the onboarding notification

If you are concerned about the less common landing pages, that would be the better approach I think. That means we don't make the request at all right? So better from a frontend and backend perspective, and for the incredibly common landing pages we continue to reuse the single request, again better for frontend and backend. 

---

On the 'scope' thing. This PR has a title "avoids data source conflict". What's the conflict we are avoiding?

Reply:
My main goal here is to avoid calling the data source that retrieves `MeshInsight`s `/meshes`. It’s too easily confused with one that retrieves `Mesh`es.

I still think we rather avoid the request to `/mesh-insights` on every load in favor of using the lighter `/meshes` endpoint.

I find it would be a practice to rely on incidental facts like "we already happen to load mesh insights on the home page so we should also use mesh insights in a component we use on every page".

Again, deciding whether we want to use another metric for whether we show the onboarding notification **is** out of scope of this PR.

Reply:
> My main goal here is to avoid calling the data source that retrieves MeshInsights /meshes. It’s too easily confused with one that retrieves Meshes.

When do we need the API response from `/meshes`?

> I still think we rather avoid the request to /mesh-insights on every load in favor of using the lighter /meshes endpoint.

We can't avoid this if we visit and/or land on the `/gui` or `/gui/meshes` pages, we need it there, but we can reuse it. If we land on deeper pages than those we can potentially avoid it entirely by checking for a 200 on deeper pages. If we do that, would that solve your problem?

Reply:
> When do we need the API response from `/meshes`?

For onboarding notifications.

> If we land on deeper pages than those we can potentially avoid it entirely by checking for a 200 on deeper pages. If we do that, would that solve your problem?

We have several pages that aren’t mesh-associated. That’s not a good substitute.

Reply:
> For onboarding notifications.

Previous to this PR, that didn't use `/meshes` and therefore currently in `master` that doesn't use `/meshes`.

Its a little strange using the change you added to justify the reason for making the change. Don't make the change, then there is no justification.


Is there any other place where we need the API response from `/meshes`?

---

> We have several pages that aren’t mesh-associated.

We have meshes (and children), and zones (and children), we also have diagnostics.

I've explained meshes and covered zones (we could reuse the zone listing response, and then use the same 200 check for anything deeper), we are then left with diagnositics. Are there any others we need to think about?

Reply:
> Previous to this PR, that didn't use `/meshes` and therefore currently in `master` that doesn't use `/meshes`.

I think I already explained how the prior change to make onboarding notification not use the /meshes API is the source of its removal and that it shouldn’t have been changed there.

So I’m not justifying my change here with something I’m introducing also but rather with the observation that it should’ve stayed like this in the first place.

> I've explained meshes and covered zones (we could reuse the zone listing response, and then use the same 200 check for anything deeper), we are then left with diagnostics. Are there any others we need to think about?

That’s a significantly more complicated logic than what we had originally which is to make a simple call to `/meshes` to find out.

Reply:
> That’s a significantly more complicated logic than what we had originally which is to make a simple call to /meshes to find out.

Maybe it is maybe it isn't, but in an ideal world this would be better for the least common usecase, yes? If it was super straight forwards to do would you prefer to do this?

Thinking about it more, we just don't show the onboarding notification on any subpages, seeing as we assume if you get a 200, then you have a mesh. That shouldn't be too difficult should it?

If you get a 404 (we are now getting into really really uncommon use case territory) we could only then call `/mesh-insights` to get the total of meshes. In fact considering how uncommon this probably is, I'd say if we get a 404 don't do anything, you have a `Go Home` button there to click on, if you click that, you'd go home, and therefore call the `/mesh-insights` endpoint (and hence share that with onboarding)

---

So:

1. If you are on a 'deep' page with a 200, just don't show the onboarding notification.
2. If you are on a deep' page and get a 404, don't do anything new, we expect the user to click the `Go Home` button
3. If you land on a higher page then we re-use `/mesh-insights`, we need to request that anyway.

^ that doesn't sound too complicated? I think its just check whether we are on a 'deep' page before showing the onboarding notification?

Reply:
This conversation was resolved through the decision to only use the onboarding notification on the home page and to use `/global-insight` data for it.

See https://github.com/kumahq/kuma-gui/pull/1746#issuecomment-1807882804